### DISCUSSION TITLE
Modeling Adjacent Compartments with "Missing" Walls

### DISCUSSION THREAD

**Discussion by anon-beaver** [firemodels/cfast#2065] [Q&A]
Hello all,

I am interested in modeling structures where the compartments are not necessarily shaped like parallelepipeds, but can be modeled as an arrangement of adjacent ones. Consider the L-shaped space in the image below, which is formed by the red and blue subspaces.

![image](https://github.com/firemodels/cfast/assets/41304722/adec19f9-3ccd-43c5-933f-2200e893497d)

Would the appropriate way to handle this case be to define the red and blue spaces as separate compartments, and then treat the "missing" wall between them with a wall vent that spans the area of the opening?

My apologies if I missed any relevant information in the user manual.

**Comment by mcgratta** [comment]
This is a common question. People have different opinions. I'll give you mine. If the blue and green spaces are not connected, I would just make the red space equal to the area of the red and blue. A fictitious vent might lead to bigger problems, as there assumptions made about vertical vents that may not be appropriate.

**Reply by drjfloyd** [reply]
The vent flow algorithm is based on the assumption that you have an orifice. An entire wall missing is not an orifice.  If it is really important that you be able to resolve different conditions between the blue and red, then CFAST may not be the tool you need.
